cpp/tests/partitioning/partition_test.cpp

```diff
@@ -15,15 +15,15 @@
  */
 #include <cudf/copying.hpp>
 #include <cudf/partitioning.hpp>
+#include <cudf/sorting.hpp>
 #include <cudf/table/table.hpp>
+
 #include <cudf_test/base_fixture.hpp>
```

Comment:
Are you trying to say that `cudf_test` goes before `cudf`? I agree with the rule, but many of our files are inconsistent about the ordering of sections. I decided to make the least disruptive change to this file in terms of lines changed, and leave that out of scope for this PR.

Reply:
Yes.
For files in src/, src is local, then include/ is farther.
For files in tests/, tests/ is local, then include/ is farther. (so `cudf_test` is closer relatively, and goes before `cudf`).
For files in benchmarks/, benchmarks/ is local, then tests/ is next, and then include/ (though include/ and tests/ are at same level, since include/cudf/ is publicly released library, I prefer to consider it to be farther).
After all these, _deps, and then conda includes, and then system std includes.

This is not in the scope of this PR. But at least on the files changed, it would be nice fix it directly. Or choose to skip it and fix include order in another PR.

Reply:
To keep this diff small and focused, I'll leave this PR as-is. Tests have already passed so I'd like to avoid running CI again. We can address this in the future, since it affects a large portion of the codebase.

Reply:
I changed my mind. I have to re-run CI for another suggestion from @PointKernel so I'll make this change now.